### ISSUE TITLE
Adding cargo slots to professions

### ISSUE THREAD

**Issue by Nightinggale** [We-the-People-civ4col-mod/Mod#141]
Medieval Conquest has the ability to add cargo slots to professions, meaning it can turn colonists into transports. I think we should do the same, hence adding an early land transport. It makes early game easier, but at the same time this kind of transport should be so bad that you don't want to use it in later game.

What I propose:
- Add a tag to xml, which sets the number of cargo slots
- Another tag sets the cargo limit for each slot
- Consider if it should have some slow feature, like can't move if it moved last turn

Another option to consider would be to add a cargo slot for food and make it revert back to NO_PROFESSION if it runs out of food. Sure this adds some micro management, but remember it's an early transport system to connect the one or two inland colonies you have before you have set up production to build proper transports.

**Comment by abmpicoli** [comment]
I always thought silly the need to have a wagon train to transport goods between colonies in the early game... I also think it would be nice to have the wagon train transport colonists at their slots...

I believe a good value for the transport would be 25 goods... or 50 if they are scouts... 

I think the food cargo slot will add too much micromanagement... unless the food is removed automatically from the city stock.... This way this will effectively add the intended effect of 

And the unit should lose hit points and eventually die if food is not found. This wouldn't happen if the unit is loaded at a transport. We could even add a promotion to units like hunters , that they replenish their food stocks (say, +2 per turn) if they have movements left at the end of the turn, and they are at a tile suitable for hunting... This would make very interesting for indians: we could trade with them for food... Making have a hit relation with haggling much more serious...

**Comment by Nightinggale** [comment]
Medieval Conquest has a different feature, which might provide the answer to the food question as well. Units can stand on a city and declare it "home city". What if whenever a unit change to transport, it stores a pointer to the city it's in and then it takes food from that city each turn? We can then restrict it to not be able to move more than x plots away from the home city.

Much less micromanagement and the units will still require food, meaning you can't exploit this to spam transports, or even worse production units should it be used for #142 as well.

**Comment by ShadesOT** [comment]
I like the idea of turning a settler into a transport. I also like the idea that settlers outside a city need food.
I think, that we should sepparate the 2 ideas.

On the topic of turning a settler into a transport, my idea is this:
I think as with all professions there are the untrained and the trained type. lets call them "carrier" and "strong carrier" for now. Each has 1 cargo slot and the trained one can carry double the amount of the untrained one. How much that is can be debated. Settlers can be turned into transports for 50 tools, 50 horses and military items of various amounts, by turning them into a pioneer, scout and military units. So that is something to think about. If that was not possible I had said 20 and 40 items. But as transporting by turning a settler into a pioneer, scout and military unit is possible, I would say ... 50 and 100? That still would not cover it entirely as a scout could move faster than an untrained carrier with 50 horses. Needs more ideas. ...

**Comment by ShadesOT** [comment]
2) on the topic of settlers needing food to survive outside of a city (or also inside a city but outside the walls):
That is a complex game feature and I am suggesting to spin it of into its own IDEA issue. That would be such a huge change to the games mechanics, that probably it would only make sense as a fork to the existing game.
So many ideas I have in my head ragarding this ... because I also thought about it.

**Comment by Nightinggale** [comment]
I opened a new ticket for the food part of the idea.

The idea is actually to have a poor transport profession, which doesn't require any yields to use. It will make inland colonies much better in early and mid game. Regardless of which path you take, getting land transport early is not easy and inland colonies becomes pointless except for land claiming.

As for how much to carry, yes we can make unitInfo contain some tag for this. We can also add a modifier to promotions, which can make sense if we start to grant free promotions though whatever mean (#93 will do wonders here).

I also have a proposal about cargo capacity in #144 where capacity becomes about total amount of yields rather than number of slots.

**Comment by LibSpit** [comment]
In M:C it actually went further and you had several carry/trader professions, so instead of building a wagon as a 'building' you made a horse drawn wagon from a colonist + X Horse + X Wood + X Tools I think. So the cost was much less about taking up building time and much more about producing the yields and using a person. 

I am not such a fan of the carrying supplies to an improvement site. It is just micro managey. In MC I think some improvements and routes took yields to build, but I think it was done through plotgroup or city radius (it took from the connected city) that and it would cost money too.

I much prefer this to the pick up, carry, repeat method as you just end up with so much back and forth. In terms of the navvy example with railroads, the material wasn't transported by the navvy, it came down the rails behind them on trains/wagons.

**Comment by Nightinggale** [comment]
I would also prefer taking yields from the nearest city and/or plotgroup. The concept of BuildInfo requiring yields could be interesting. This would require building a road and then upgrade to railroad purely because the road would add the plot to a plotgroup.

`
In terms of the navvy example with railroads, the material wasn't transported by the navvy, it came down the rails behind them on trains/wagons.
`
In America railroads often started in one end (or both in one famous example) and then they had the concept of a railhead, which is where construction took place. A railhead would always be connected by rail to the starting point, meaning it could be supplied by trains. Europe on the other hand had a well developed road network prior to the railroads and had much easier access to working on multiple locations at once if it was judged to be beneficial. There is one more supply option: navvies producing items at location. Sleepers were treated to make them last longer and this treatment was often done by the navies.

**Comment by LibSpit** [comment]
Yeah, my basic point was that generally builders don't carry the materials, someone does that for them, the cost can then be abstracted to money or 'local' yields. It gives the desired 'maintenance' of your empire without having to manually fiddle everything.

**Comment by dcrookston** [comment]
> I think as with all professions there are the untrained and the trained type. lets call them "carrier" and "strong carrier" for now. Each has 1 cargo slot and the trained one can carry double the amount of the untrained one.

Why not just make everyone able to carry 20 goods, Hardy Pioneers and scouts can carry 40, and Seasoned Scouts (if they are on horseback only!) can carry 60?  This eliminates the need for adding a "carrier" profession that will be useless as soon as you get wagons, and it sort of fits with what these units do anyway.  Also it would make seasoned scouts actually useful when there's no more exploring to do.

**Comment by ErikSandbraaten** [comment]
In my opinion, adding cargo slots to native units (or perhaps creating a new native "peddler\"packmule" profession) would neatly solve the cheat where the natives get to "teleport" yields between settlements\units.

**Comment by Drunk-Monk** [comment]
> In my opinion, adding cargo slots to native units (or perhaps creating a new native "peddler and\or "packmule" profession) would neatly solve the cheat where the natives get to "teleport" yields between settlements\units.
Slightly off-topic, but I also want natives to have a coastal transport as well, something like a "canoe" and\or "raft" unit.

Both of these are really good ideas, some native groups used water transport extensively and overall the native factions need more depth to them.

In general whilst the concept of a unit being able to carry goods is sound, I feel it needs to be limited to specific units like the scout to stop it being exploitable.
